src/test/java/com/treasuredata/client/TestTDClient.java

```diff
@@ -1958,6 +1959,50 @@ public void testImportBytesWithId()
         assertEquals(result.getUniqueId(), "4288048cf8f811e88b560a87157ac806");
     }
 
+    @Test
+    public void testGetFederatedQueryConfigsWhenEmpty()
+        throws Exception
+    {
+        client = mockClient();
+        server.enqueue(new MockResponse().setBody("[]"));
+
+        List<TDFederatedQueryConfig> result = client.getFederatedQueryConfigs();
+
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void testGetFederatedQueryConfigsWhenOnlyOne()
+        throws Exception
+    {
+        client = mockClient();
+        final String federatedQueryConfigs = "[{\"id\":12345,\"type\":\"custom_type\",\"user_id\":67890,\"account_id\":54321,\"name\":\"test_name\",\"connection_id\":112233,\"created_at\":\"2023-06-01T00:00:00Z\",\"updated_at\":\"2023-06-15T00:00:00Z\",\"settings\":{\"account_name\":\"snowflake_account\",\"auth_method\":\"password\",\"private_key\":\"encrypted_key\",\"passphrase\":\"secret_phrase\",\"options\":{\"key1\":\"val1\",\"key2\":\"val2\"},\"user\":\"snowflake_user\",\"password\":\"snowflake_password\",\"database\":\"sample_database\"}}]";
+        server.enqueue(new MockResponse().setBody(federatedQueryConfigs));
+
+        List<TDFederatedQueryConfig> result = client.getFederatedQueryConfigs();
+
+        List<TDFederatedQueryConfig> expected = Arrays.asList(new TDFederatedQueryConfig(12345, "custom_type", 67890, 54321, "test_name", 112233, "2023-06-01T00:00:00Z", "2023-06-15T00:00:00Z", "{\"account_name\":\"snowflake_account\",\"auth_method\":\"password\",\"private_key\":\"encrypted_key\",\"passphrase\":\"secret_phrase\",\"options\":{\"key1\":\"val1\",\"key2\":\"val2\"},\"user\":\"snowflake_user\",\"password\":\"snowflake_password\",\"database\":\"sample_database\"}"));
+        assertEquals(result, expected);
```

Comment:
`Assertions.assertIterableEquals` should be used.
```suggestion
        assertIterableEquals(result, expected);
```